### DISCUSSION TITLE
How to use CTS RTS

### DISCUSSION THREAD

**Discussion by utya** [stm32duino/Arduino_Core_STM32#1681] [General]
Hello. My gps uses CTS/RTS pin. 
Im using Stm32L476JG, how i can enable cts/rts. The gps connected to USART. I tied
HardwareSerial Serial1(PA_10, PA_9);

Serial1.setCts(PB_4);
Serial1.setRts(PG_12);

please help

**Comment by ABOSTM** [comment]
HI @utya,
I don't see obvious reason for CTS/RTS not working,
pin configuration seems correct.

Knowing that variant Stm32L476JG doesn't yet exist in this core, which variant did you used ?
Note:  you can add this new variant following this wiki: https://github.com/stm32duino/wiki/wiki/Add-a-new-variant-%28board%29
especially the `3 - Generic System Clock configuration`, which may impact the USART capability

**Comment by utya** [comment]
@ABOSTM 
thanks for replay. Im using nucleo-L476RQ (variant). I don't maybe it's not correct.
i could check another uart, and it worked with manually configured pins like this 
HarwareSerial serial5 (PD_2, PC_12).
rts/cts work if download release by git, not by Arduino Broad manager

**Reply by fpistm** [reply]
> rts/cts work if download release by git, not by Arduino Broad manager

This is normal as this feature has been added recently and will be available within the next release. See #1634

**Comment by ABOSTM** [comment]
nucleo-L476RQ doesn't exist, I guess you mean nucleo-L476RG.
nucleo-L476RG doesn't have PG_12 so you cannot use nucleo-L476RG variant to use  PG_12 as RTS for USART : 
It is not in the peripheral pin array PinMap_UART_RTS[] for this board
https://github.com/stm32duino/Arduino_Core_STM32/blob/e156c32db24d69cb4818208ccc28894e2f427cfa/variants/STM32L4xx/L475R(C-E-G)T_L476R(C-E-G)T_L486RGT/PeripheralPins.c#L215

But PG_12 is defined in PinMap_UART_RTS[] for Stm32L476JG https://github.com/stm32duino/Arduino_Core_STM32/blob/e156c32db24d69cb4818208ccc28894e2f427cfa/variants/STM32L4xx/L476J(E-G)Y_L485J(C-E)Y_L486JGY/PeripheralPins.c#L232

So I strongly suggest to follow the wiki to add your variant:
https://github.com/stm32duino/wiki/wiki/Add-a-new-variant-%28board%29

**Reply by utya** [reply]
i understood, thanks i will try to add by wiki. but unfortunately I don’t know what kind of quartz I have on the board. 
if the configuration from the nucleo worked, then  i should to copy it from there

nucleo-L476RQ doesn't exist, I guess you mean nucleo-L476RG.
yes. it's true.
